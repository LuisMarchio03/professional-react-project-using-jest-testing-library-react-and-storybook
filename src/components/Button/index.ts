import { styled } from '../../styles'

export const Button = styled('button', {
  fontSize: '$md',
  fontWeight: 600,
  textTransform: 'uppercase',
  padding: '12px 36px',
  cursor: 'pointer',

  variants: {
    variant: {
      primary: {
        backgroundColor: '$yellow500',
        border: '2px solid $yellow500',
        color: '$gray600',
      },
      secondary: {
        backgroundColor: 'transparent',
        border: '2px solid $yellow500',
        color: '$yellow500',
      },
    },
  },
})
