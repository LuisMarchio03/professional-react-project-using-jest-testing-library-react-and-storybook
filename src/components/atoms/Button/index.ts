import { styled } from '../../../styles'

export const Button = styled('button', {
  fontSize: '$md',
  fontWeight: 600,
  textTransform: 'uppercase',
  padding: '12px 36px',
  cursor: 'pointer',

  transition: '0.2s',

  variants: {
    variant: {
      primary: {
        backgroundColor: '$yellow500',
        border: '2px solid $yellow500',
        color: '$gray600',

        '&:hover:enabled': {
          backgroundColor: '$yellow700',
        },

        '&:disabled': {
          opacity: 0.7,
          backgroundColor: '$yellow700',
          cursor: 'not-allowed',
        },
      },
      secondary: {
        backgroundColor: 'transparent',
        border: '2px solid $yellow500',
        color: '$yellow500',

        '&:hover:enabled': {
          color: '$yellow700',
          border: '2px solid $yellow700',
        },

        '&:disabled': {
          opacity: 0.7,
          color: '$yellow700',
          border: '2px solid $yellow700',
          cursor: 'not-allowed',
        },
      },
      danger: {
        backgroundColor: 'transparent',
        border: '2px solid $red500',
        color: '$red500',

        '&:hover:enabled': {
          color: '$red700',
          border: '2px solid $red700',
        },

        '&:disabled': {
          opacity: 0.7,
          color: '$red700',
          border: '2px solid $red700',
          cursor: 'not-allowed',
        },
      },
    },
  },
})
