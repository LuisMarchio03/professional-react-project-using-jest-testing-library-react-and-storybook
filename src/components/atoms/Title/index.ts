import { styled } from '../../../styles'

export const Title = styled('h1', {
  position: 'relative',
  // fontWeight: 700,
  letterSpacing: '0.05rem',
  marginBottom: '25px',
  paddingBottom: '25px',

  borderBottom: '1px solid rgba(255,255,255,0.2)',

  '&::after': {
    content: '""',
    position: 'absolute',
    left: '0',
    bottom: '-3px',
    backgroundColor: '$yellow500',
    width: '70px',
    height: '5px',
  },

  strong: {
    color: '$yellow500',
  },
})
