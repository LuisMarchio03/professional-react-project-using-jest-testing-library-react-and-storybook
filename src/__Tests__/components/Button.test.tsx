import React from 'react'
import { fireEvent, render } from '@testing-library/react'

import { Button } from '../../components/atoms/Button'

test('renders a text', () => {
  const { getByText } = render(<Button>Click here</Button>)
  expect(getByText('Click here')).toBeInTheDocument()
})

test('trigger event on click', () => {
  const handleClick = jest.fn()
  const { getByRole } = render(<Button onClick={handleClick} />)

  fireEvent.click(getByRole('button'))

  expect(handleClick).toBeCalled()
})

test('should not be able to trigger event on click with button disabled', () => {
  const handleClick = jest.fn()
  const { getByRole } = render(<Button disabled onClick={handleClick} />)

  fireEvent.click(getByRole('button'))

  expect(handleClick).not.toBeCalled()
})

test('[Primary] variants button', () => {
  const { asFragment } = render(<Button variant="primary" />)
  expect(asFragment()).toMatchSnapshot()
})

test('[Secondary] variants button', () => {
  const { asFragment } = render(<Button variant="secondary" />)
  expect(asFragment()).toMatchSnapshot()
})

test('[Danger] variants button', () => {
  const { asFragment } = render(<Button variant="danger" />)
  expect(asFragment()).toMatchSnapshot()
})
