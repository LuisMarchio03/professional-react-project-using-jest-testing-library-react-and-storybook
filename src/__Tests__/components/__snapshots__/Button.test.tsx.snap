// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Danger] variants button 1`] = `
<DocumentFragment>
  <button
    class="c-jGmEde c-jGmEde-dIixzL-variant-danger"
  />
</DocumentFragment>
`;

exports[`[Primary] variants button 1`] = `
<DocumentFragment>
  <button
    class="c-jGmEde c-jGmEde-kaJcgy-variant-primary"
  />
</DocumentFragment>
`;

exports[`[Secondary] variants button 1`] = `
<DocumentFragment>
  <button
    class="c-jGmEde c-jGmEde-jOKMJT-variant-secondary"
  />
</DocumentFragment>
`;
